Update tests to go1.16.2

andygabby opened the pull request:
Update testing from go1.16.1 to 1.16.2

Fixes #5333

beautifulentropy reviewed (changes requested):
1. If we have already been testing with go `1.16.1` then this should be `1.16.1_<date>` and `1.16.2_<date>`
2. the `go` key of `.travis.yml` needs to be updated
3. the `image` key of `docker-compose.yml` needs to be updated

andygabby commented:
> * the `go` key of `.travis.yml` needs to be updated

When we moved to github actions, the code changes broke .travis.yml golang selector. Currently .travis.yml just falls back to the default in docker-compose.yml. There was some consensus that it was unnecessary to add the complexity to make both work in parallel when we would remove travis on the heels of adding github actions.

> * the `image` key of `docker-compose.yml` needs to be updated

I have updated docker-compose.yml to use the newly tagged default 1.15.7. Functionally it is the exact same image as the previous, but I do like to keep the date tags the same on the two different versions of go we are testing.
